ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -0,0 +1,233 @@
+    private static final String BASE_URL = "https://stage.cru.org";
+
+    /**
+     * Constructor
+     * @param jsonObject = the Object to be parsed.
+     */
+    public ArticleParser(JSONObject jsonObject) {
+        this.articleJSON = jsonObject;
+    }
+
+    //region Public Executor
+    /**
+     * This execute function allows you to change the initial jsonObject and run the execution.
+     *
+     * use <code>ARTICLE_LIST_KEY</code> and <code>ATTACHMENT_LIST_KEY</code> to access the Collections
+     * out of the HashMaps
+     *
+     * @param jsonObject = the object to be parsed.
+     * @return = return a list of <code>Article</code> and <code>Attachments</code> in HashMap
+     */
+    public HashMap<String, Object> execute(JSONObject jsonObject) {
+        this.articleJSON = jsonObject;
+        return execute();
+    }
+
+    /**
+     * This executes the parsing of the local JsonObject.
+     *
+     * use <code>ARTICLE_LIST_KEY</code> and <code>ATTACHMENT_LIST_KEY</code> to access the Collections
+     * out of the HashMaps
+     *
+     * @return = return a list of <code>Article</code> and <code>Attachments</code> in HashMap
+     */
+    public HashMap<String, Object> execute() {
+
+        jsonObjectHandler(articleJSON);
+
+        HashMap<String, Object> returnObject = new HashMap<>();
+        returnObject.put(ARTICLE_LIST_KEY, articleList);
+        returnObject.put(ATTACHMENT_LIST_KEY, attachmentList);
+
+        return returnObject;
+    }
+    //endregion public Executor
+
+    //region Article Parsing
+
+    /**
+     * This method takes in a Json Object and Iterates through the keys to determine if they
+     * are Article Objects or Category Object.  Article Objects will be passed on to be parsed
+     * and Category Object will be Recursively placed back into this method.
+     * @param evaluatingObject = the Json Object to be evaluated.
+     */
+    private void jsonObjectHandler(JSONObject evaluatingObject) {
+        // Create loop through Keys
+        Iterator<String> keys = evaluatingObject.keys();
+        while (keys.hasNext()) {
+            String nextKey = keys.next();
+            if (isArticleOrCategory(nextKey)) {
+                try {
+                    JSONObject returnedObject = evaluatingObject.getJSONObject(nextKey);
+                    if (isObjectCategory(returnedObject)) {
+                        jsonObjectHandler(returnedObject);
+                    } else {
+                        // Get inner Article Object
+                        parseArticleObject(returnedObject);
+                    }
+                } catch (Exception e) {
+                    Timber.e(e, "getArticleFromCategory: ");
+                }
+            }
+        }
+    }
+
+    /**
+     * This method parses an Article Json Object into the Database. On completion it will add
+     * articles to <code>articleList</code>
+     *
+     * @param articleObject = article JsonObject
```

Comment:
the actual javadoc format for parameters doesn't use an `=`, it should just be:
```
* @param articleObject article JsonObject
```
example: https://www.oracle.com/technetwork/articles/java/index-137868.html#format